subprojects/base-services/src/main/java/org/gradle/internal/hash/Hashing.java

```diff
@@ -199,18 +198,14 @@ protected MessageDigest createDigest() {
     }
 
     private static class MessageDigestHasher implements PrimitiveHasher {
-        private final MessageDigest digest;
-        private final ByteBuffer buffer = ByteBuffer.allocate(8).order(ByteOrder.LITTLE_ENDIAN);
-        private boolean done;
```

Comment:
instead of a boolean value, we're `null`-ing out `digest` at the end to make sure it's not reused